### PR TITLE
[Event Hubs Client] September Release Preparation (Core)

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to prepare the core `Azure.Messaging.EventHubs` library for its September release.

# Last Upstream Rebase

Saturday, September 5, 10:02am (EDT)

### CODE REVIEW

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -9,11 +9,12 @@
   <ItemGroup>
     <PackageReference Update="ApprovalTests" Version="3.0.22" />
     <PackageReference Update="ApprovalUtilities" Version="3.0.22" />
-    <PackageReference Update="Azure.AI.TextAnalytics" Version="1.0.0" />
+    <PackageReference Update="Azure.AI.FormRecognizer" Version="3.0.0" />
```
Comment (jsquire) [inline eng/Packages.Data.props:12]: These changes are being sync'd from `master`, based on commit `0abf2a2ee7104bc0922775373f34aa197554a95b`, reflecting the `Azure.Core` release.   

I've confirmed none of the changes will impact the Event Hubs release in an unexpected or negative way.

**Comment by jsquire** [comment]
Once committed to this release branch, I'll cherry-pick and merge to `master` in a separate PR.

File: eng/Packages.Data.props (hunk 1 above)
Comment (jsquire) [reply eng/Packages.Data.props]: These were causing conflicts elsewhere in the tree due to changes with the `BinaryData` type in `Azure.Core.Experimental`;  I've reverted this and used local version overrides for Event Hubs to avoid orthogonal updates elsewhere.

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj [hunk 2]
```diff
@@ -13,7 +13,11 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Messaging.EventHubs" VersionOverride="5.2.0-preview.3" /> <!--Version override exists to bind to the preview; this will be removed when v5.2.0 is released for GA -->
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Azure.Messaging.EventHubs.Processor.csproj:15]: This reference and the project version will be updated for release in a separate PR, once the `Azure.Messaging.EventHubs` package for September has been published.

**Review by christothes — APPROVED** [review]
